### ISSUE TITLE
Support configuring `WP_HOME` with alternate port

### ISSUE THREAD

**Issue by joemcgill** [WordPress/gutenberg#26481]
**Is your feature request related to a problem? Please describe.**
I want to use `@wordpress/env` to create an environment to test a headless WordPress application where the front end will be run from a node application running outside of the Env's docker setup. To do this, I want to be able to specify a `WP_HOME` like `http://localhost:3001` in my `.wp-env.json` file and have the port specified be respected.

However, right now the `withOverrides()` function in [config/config.js](https://github.com/WordPress/gutenberg/blob/master/packages/env/lib/config/config.js#L240-L278) overrides the port with the port of the environment specified.

**Describe the solution you'd like**
I'd like to see `withOverrides()` check to see if a port value is set for `WP_HOME` before overriding it here: https://github.com/WordPress/gutenberg/blob/master/packages/env/lib/config/config.js#L262 I think overriding the `WP_TESTS_DOMAIN` and `WP_SITEURL` probably makes sense to leave as is.

**Describe alternatives you've considered**

**Comment by noahtallen** [comment]
@joemcgill I'm not super familiar with how these settings are often used. I coded them to be the same because I had some trouble when they were different :) Is there a reason why using the same port for each of these URLs won't work?

What kind of use case would this enable, out of curiosity?

**Comment by joemcgill** [comment]
Hi @noahshrader, like I described in the issue description, I'm wanting to use wp-env run the backend for a headless site where `WP_HOME` points to a separate node application. Right now, if I have a headless front end running on a different port than the WP container and attempt to set `WP_HOME` to that application (i.e., the front end should be localhost:3030 and wp-env is on localhost:8888) then this function strips off the port that I supplied in my `.wp-env.json` file and replaces it with the port for the WP container (i.e. 8888).
